src/libraries/System.Diagnostics.Process/src/System/Diagnostics/ProcessManager.Win32.cs

```diff
@@ -330,71 +330,63 @@ private static unsafe ProcessInfo[] GetProcessInfos(ReadOnlySpan<byte> data, int
                 ref readonly SYSTEM_PROCESS_INFORMATION pi = ref MemoryMarshal.AsRef<SYSTEM_PROCESS_INFORMATION>(data.Slice(processInformationOffset));
 
                 // Process ID shouldn't overflow. OS API GetCurrentProcessID returns DWORD.
-                int processInfoProcessId = pi.UniqueProcessId.ToInt32();
-                if (processIdFilter == null || processIdFilter.GetValueOrDefault() == processInfoProcessId)
+                int processId = pi.UniqueProcessId.ToInt32();
+                if (processIdFilter == null || processIdFilter.GetValueOrDefault() == processId)
                 {
-                    // get information for a process
-                    ProcessInfo processInfo = new ProcessInfo((int)pi.NumberOfThreads)
+                    string? processName = null;
+                    ReadOnlySpan<char> processNameSpan =
+                        pi.ImageName.Buffer != IntPtr.Zero ? GetProcessShortName(new ReadOnlySpan<char>(pi.ImageName.Buffer.ToPointer(), pi.ImageName.Length / sizeof(char))) :
+                        (processName =
+                            processId == NtProcessManager.SystemProcessID ? "System" :
+                            processId == NtProcessManager.IdleProcessID ? "Idle" :
+                            processId.ToString(CultureInfo.InvariantCulture)); // use the process ID for a normal process without a name
+
+                    if (string.IsNullOrEmpty(processNameFilter) || processNameSpan.SequenceEqual(processNameFilter))
```

Comment:
should this be using case insensitive comparison?

Reply:
Added